pkg/kubelet/dockershim/security_context.go

```diff
@@ -113,6 +113,14 @@ func modifyHostConfig(sc *runtimeapi.LinuxContainerSecurityContext, hostConfig *
 	}
 	hostConfig.SecurityOpt = append(hostConfig.SecurityOpt, apparmorSecurityOpts...)
 
+	if sc.NoNewPrivs {
+		// "no-new-privileges" was added in Docker 1.11
+		// which is API version 1.23
+		if apiVersion.GTE(semver.Version{Major: 1, Minor: 23}) {
```

Comment:
It doesn't need to be perfect... You could start with `runtime == rkt || (runtime == docker && apiversion >= 1.23)`